modules/template/src/main/twirl/scaladex/view/project/artifacts.scala.html

```diff
@@ -10,103 +10,92 @@
+                      </div>
+                      <div class="col-md-9">
+                        <div class="artifact-info">
+                          <a data-toggle="tooltip" data-placement="top" title="@formatInstant(artifacts.head.releaseDate)">
+                            @TimeUtils.toFiniteDuration(artifacts.head.releaseDate, Instant.now()).prettyPrint
+                            ago<i class="fa-solid fa-calendar-days"></i>
+                          </a>
+                          <br>
+                          @artifacts.head.licenses.map { license =>
+                            <a href="@license.url.getOrElse("#")">@license.shortName<i class="fa-solid fa-scale-balanced"></i></a>
+                          }
+                        </div>
+                        <div class="artifact-content">
+                          @for(
+                            (platform, binaryVersions) <- artifacts
+                              .groupBy(_.platform)
+                              .toSeq
+                              .sortBy(_._1)
+                              .reverse
+                          ) {
+                            <h5>@platform:</h5>
+                            @for(binaryVersion <- binaryVersions.map(_.binaryVersion).distinct.sorted.reverse) { 
+                              <a href="/@project.reference/artifacts/@artifactName/@version?binary-version=@binaryVersion.encode" class="scala-version">
+                                @binaryVersion.language.label
+                              </a>
+                            }
+                            <br>
+                          }
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                }
+              }
+            }
           </div>
-          }
         }
-        </div>
       </div>
     </div>
   </main>
 }
-
 @filters() = {
-  <form action="#" action="GET">
-    <select class="selectpicker" data-style="btn-primary" onchange="window.location=this.value">
-      @for(name <- header.allArtifactNames) {
-          <option value="@artifactsUri(project.reference, name, params)"
-          @if(name == artifactName) {selected}>
-          @name
-          </option>
-      }
-    </select>
-
+  <form action="#" action="GET"> 
     <select class="selectpicker" name="binary-versions" 
-      title= "Scala Versions" multiple data-style="btn-primary" data-actions-box="true" data-selected-text-format="static"
+      title="Scala Versions" multiple data-style="btn-primary" data-actions-box="true" data-selected-text-format="static"
       onchange="this.form.submit()">
-      @for((platform, binaryVersions) <- allBinaryVersions.sorted.groupBy(_.platform)) {
+      @for((platform, binaryVersions) <- groupedArtifacts.values.flatten.flatMap(_._2).map(_.binaryVersion).toSeq.distinct.groupBy(_.platform).toSeq.sortBy(_._1).reverse) {
```

Comment:
This is really hard to read. I suggest you pass an `allBinaryVersions` as parameter of this page, and you compute it in `ProjectPages` instead.